modules/core/src/test/java/org/apache/ignite/logger/java/JavaLoggerTest.java

```diff
@@ -84,4 +207,13 @@ public void testLogInitialize() throws Exception {
         assert !log.fileName().contains("%");
         assert log.fileName().contains("other-app");
     }
+
+    /**
+     * Configure java.util.logging.config.file property
+     */
+    private void setJavaLoggerConfig() throws IOException {
+        File file = new File(U.getIgniteHome(), LOG_CONFIG_DEBUG);
+        System.setProperty("java.util.logging.config.file", file.getPath());
+        LogManager.getLogManager().readConfiguration();
```

Comment:
I suppose user shouldn't explicitly read configuration. It is invoked inside LogManager while user tries to log first message, isn't it?

Reply:
The problem is that LogManager reads the property when `Logger.getLogger` method is called.

Firstly, it happens because JavaLoggerTest extends GridCommonAbstractTest which initialize IgniteUtils and IgniteUtils calls Logger.getLogger

Secondly, U.getIgniteHome() also calls initialization of IgniteUtils and it calls `Logger.getLogger` again.

So `Logger.getLogger` is called before setProperty and that's why jul won't read this property without explicit configuration.